### ISSUE TITLE
Model creation error in 

### ISSUE THREAD

**Issue by dakab1** [jazzband/django-authority#65]
Hi Guys,

In Django version 2.1.7 and Authority 0.13.2 I get the below error:

### The problem

```
  File "/home/dean/Documents/mycio/mycio-simafrica/mycio-simafrica-env/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/dean/Documents/mycio/mycio-simafrica/mycio-simafrica-env/lib/python3.7/site-packages/authority/migrations/0001_initial.py", line 9, in <module>
    class Migration(migrations.Migration):
  File "/home/dean/Documents/mycio/mycio-simafrica/mycio-simafrica-env/lib/python3.7/site-packages/authority/migrations/0001_initial.py", line 30, in Migration
    ('user', models.ForeignKey(related_name='granted_permissions', blank=True, to=settings.AUTH_USER_MODEL, null=True)),
TypeError: __init__() missing 1 required positional argument: 'on_delete'

```

### The fix
I was able to fix in the following files:
#### models.py:
Added , ondelete.PROTECT to all ForeignKey fields as follows
```
    content_type = models.ForeignKey(ContentType, related_name="row_permissions", on_delete=models.PROTECT )
    user = models.ForeignKey(
        user_model_label, null=True, blank=True, related_name='granted_permissions', on_delete=models.PROTECT)
    group = models.ForeignKey(Group, null=True, blank=True, on_delete=models.PROTECT)
    creator = models.ForeignKey(
        user_model_label, null=True, blank=True, related_name='created_permissions', on_delete=models.PROTECT)

```
I also had to update the 00001_initial.py file, but this one could be a per machine thing because that script might have been generated the first time i initialized authority code from my application.
#### 00001_initial.py:
Also added , ondelete.PROTECT to all ForeignKey fields as follows:

```
                ('content_type', models.ForeignKey(related_name='row_permissions', to='contenttypes.ContentType', on_delete=models.PROTECT)),
                ('creator', models.ForeignKey(related_name='created_permissions', blank=True, to=settings.AUTH_USER_MODEL, null=True, on_delete=models.PROTECT)),
                ('group', models.ForeignKey(blank=True, to='auth.Group', null=True, on_delete=models.PROTECT)),
                ('user', models.ForeignKey(related_name='granted_permissions', blank=True, to=settings.AUTH_USER_MODEL, null=True, on_delete=models.PROTECT)),

```

Not sure if i should fix, branch out and send a pull request or if this was fixed in another version?

**Comment by jlward** [comment]
I have a fix for this in #64 . For some reason, I can't get passing tests with python3 on travis (they pass locally with tox). I opted to use `models.CASCADE` in favor of `models.PROTECT` because then we are not changing functionality. If we switch to `PROTECT`, we'd first have to update all migrations to be `CASCADE`, then make a new migration for switching to `PROTECT`. Otherwise, existing apps would have to run the migration manually to get protect to happen. 

I also thing that if a user or group is deleted, we should not be protecting that user/group if they have permissions.

**Comment by dakab1** [comment]
Thanks for the response @jlward . So do you recommend I pull from master and that should be compatible with Django version 2+

**Comment by jlward** [comment]
> Thanks for the response @jlward . So do you recommend I pull from master and that should be compatible with Django version 2+ ?

We haven't gotten #64 merged in yet. So master is still busted in Django 2+.

**Comment by jezdez** [comment]
This is merged now and will be part of 0.14.
